Compfiles/Imo1970P4.lean

```diff
@@ -21,15 +20,781 @@ equals the product of the nubmers in the other set.
+            dsimp[Odd]
+            dsimp[Odd] at h
+            obtain ⟨k, h6⟩ := h
+            use k + 1
+            rw[h6]
+            ring_nf
+          case inr h6 =>
+            simp_all only
+            dsimp[Odd]
+            dsimp[Odd] at h
+            obtain ⟨k, h6⟩ := h
+            use k + 2
+            rw[h6]
+            ring_nf
+      intro H
+      obtain ⟨a, Hh⟩ := H
+      have h3 := Odd.not_two_dvd_nat Hh
+      by_contra Hhh
+      simp_all only [Nat.two_dvd_ne_zero, not_or]
+      omega
+  case inr h =>
+    use n + 1
+    use n + 3
+    use n + 5
+    simp_all
+    have := Even.two_dvd h
+    ext x
+    simp_all
+    apply Iff.intro
+    intro H
+    constructor
+    · omega
+    · cases H
+      case inl h3 =>
+        simp_all only [Even.add_one]
+      case inr h4 =>
+        cases h4
+        case inl h5 =>
+          have : Odd 3 := by decide
+          have := Even.add_odd h this
+          rw[h5]
+          exact this
+        case inr h6 =>
+          have : Odd 5 := by decide
+          have := Even.add_odd h this
+          rw[h6]
+          exact this
+    intro H
+    obtain ⟨a, Hh⟩ := H
+    have h3 := Odd.not_two_dvd_nat Hh
+    by_contra Hhh
+    simp_all only [Nat.two_dvd_ne_zero, not_or]
+    omega
+
+lemma exactly_three_odd_numbers (n : ℕ) (odd_s : Finset ℕ)
+                                (odd_s_eq: odd_s = (Finset.Icc n (n + 5)).filter (λ x => Odd x)): (odd_s).card = 3 := by
+  -- ∃ (a b c : ℕ), {a, b, c} = odd_s ∧ odd_s.card = 3
+  obtain ⟨x, y, z, ⟨left, ⟨y_eq, z_eq⟩⟩⟩ := odd_props n odd_s odd_s_eq
+  have := (@Finset.card_eq_three ℕ odd_s).mpr
+  apply this
+  use x
+  use x + 2
+  use x + 2 + 2
+  constructor
+  · omega
+  · constructor
+    · omega
+    · constructor
+      · omega
+      · simp_all only
+
+
+
+
+
+
+lemma at_most_one (n : ℕ) (x y : ℕ)
+  (x_in_interval : x ∈ Finset.Icc n (n + 5)) (y_in_interval : y ∈ Finset.Icc n (n + 5))
+  (x_div_5 : 5 ∣ x )
+  (x_non_div_2 : ¬ 2 ∣ x)
+  (y_div_5 : 5 ∣ y )
+  (y_non_div_2 : ¬ 2 ∣ y) :
+  x = y := by
+  simp_all only [Finset.mem_Icc, Nat.two_dvd_ne_zero]
+  omega
+
+lemma card_of_equal (s : Finset ℕ) : (∀ x ∈ s, ∀ y ∈ s,  x = y) → s.card ≤ 1 := by
+  exact (@Finset.card_le_one ℕ s).mpr
+
+lemma five_divides_odd_at_most_once (n : ℕ) (s odd_s : Finset ℕ) (partition : s = Finset.Icc n (n + 5))
+                                    (odd_s_eq: odd_s = s.filter (λ x => Odd x)) : (odd_s.filter (λ x => 5 ∣ x)).card ≤ 1 := by
+  have : (∀ x ∈ (odd_s.filter (λ x => 5 ∣ x)), ∀ y ∈ (odd_s.filter (λ y => 5 ∣ y)), x = y) → (odd_s.filter (λ x => 5 ∣ x)).card ≤ 1 := by
+    exact (card_of_equal) (odd_s.filter (λ x => 5 ∣ x))
+
+  apply this
+
+  rw[odd_s_eq]
+
+  intro x x_in
+  intro y y_in
+
+  simp only [Finset.mem_filter] at x_in
+  obtain ⟨⟨X1, X2⟩, X3⟩ := x_in
+
+  simp only [Finset.mem_filter] at y_in
+  obtain ⟨⟨Y1, Y2⟩, Y3⟩ := y_in
+
+  apply at_most_one n
+  rw[partition] at X1
+  exact X1
+  rw[partition] at Y1
+  exact Y1
+  exact X3
```

Comment:
These successive `exact`s indicate that the proof needs to be structured via `·` and indentation, so that it focuses on one goal at time. Looks like these subgoals were introduced after `apply at_most_one n`. You should add cdot and indentation there.